File: fleeman/src/main/java/com/example/demo/entities/Hub.java
package com.example.demo.entities;
import jakarta.persistence.*;

@Entity
@Table(name = "Hub")
public class Hub {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int hubId;
    
    private String hubName;
    
    @Column(name = "Hub_Address_and_Details", columnDefinition = "TEXT")
    private String hubAddressAndDetails;
    
    @Column(name = "contactNumber", unique = true) 
    private Long contactNumber;
    
    @ManyToOne
    @JoinColumn(name = "city_id")
    private City city;
    
    @ManyToOne
    @JoinColumn(name = "state_id")
    private State state;

    // Constructors
    
    public Hub() {
    }

    // Getters and setters

    public int getHubId() {
        return hubId;
    }

    public void setHubId(int hubId) {
        this.hubId = hubId;
    }

    public String getHubName() {
        return hubName;
    }

    public void setHubName(String hubName) {
        this.hubName = hubName;
    }

    public String getHubAddressAndDetails() {
        return hubAddressAndDetails;
    }

    public void setHubAddressAndDetails(String hubAddressAndDetails) {
        this.hubAddressAndDetails = hubAddressAndDetails;
    }

    public Long getContactNumber() {
        return contactNumber;
    }

    public void setContactNumber(Long contactNumber) {
        this.contactNumber = contactNumber;
    }

    public City getCity() {
        return city;
    }

    public void setCity(City city) {
        this.city = city;
    }

    public State getState() {
        return state;
    }

    public void setState(State state) {
        this.state = state;
    }

    // toString method

    @Override
    public String toString() {
        return "Hub{" +
                "hubId=" + hubId +
                ", hubName='" + hubName + '\'' +
                ", hubAddressAndDetails='" + hubAddressAndDetails + '\'' +
                ", contactNumber=" + contactNumber +
                ", city=" + city +
                ", state=" + state +
                '}';
    }
}

File: pro3/Models/Hub.cs
using Microsoft.EntityFrameworkCore.Migrations;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace pro3.Models
{
    public class Hub
    {
        [Key]
        [Column(name: "Hub_Id")]
        public int? HubId { get; set; }

        [Required]
        [Column(name: "Hub_Name")]
        public string HubName { get; set; }

        [Column(name: "Hub_Address_and_Details", TypeName = "text")]
        public string HubAddressAndDetails { get; set; }

        [Column(name: "Contact_Number")]
        public long ContactNumber { get; set; }

        [ForeignKey("City")]
        [Column(name: "City_Id")]
        public int? CityId { get; set; }

        [ForeignKey("State")]
        [Column(name: "State_Id")]
        public int? StateId { get; set; }

        public virtual City? City { get; set; }

        public virtual State? State { get; set; }
    }
}
